torch/csrc/autograd/engine.cpp

```diff
@@ -93,6 +93,23 @@ struct ReadyQueue {
   size_t size() const;
 };
 
+// This shared_ptr is a thread local pointer to the local ready queue per thread
```

Comment:
I found this comment pretty confusing on the initial read. I realize it references `Note [Allocating GPUs to autograd threads]` but I think it would be super helpful for context to just say, first, "Every autograd worker thread is associated with a ready queue, which specifies the stream of work of this thread to do." The difference between this sentence and what's written here is that this sentence explains the "Why", whereas what's written here just says "What" (you nearly get there when you say "used as the queue for executing tasks", but this elocution is pretty vague!)

The later description talks about Engine "memorizing" the local ready queues, but I feel it would probably be easier to understand if you discussed briefly how work gets allocated to threads (the CUDA threads are shared amongst all invocations of backwards, while CPU threads are dedicated to processing CPU work for the backward they invoked). Then "obviously" any given graph task maintains where you should send work for it to be done.